test/1_stdlib/subString.swift

```diff
@@ -1,15 +1,110 @@
-// RUN: %target-run-simple-swift | FileCheck %s
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: %target-build-swift %s -o %t/a.out
+// RUN: %target-run %t/a.out
 // REQUIRES: executable_test
 
-func test(_ s: String) {
-  print(s)
-  var s2 = s[s.index(s.startIndex, offsetBy: 2)..<s.index(s.startIndex, offsetBy: 4)]
-  print(s2)
-  var s3 = s2[s2.startIndex..<s2.startIndex]
-  var s4 = s3[s2.startIndex..<s2.startIndex]
+import StdlibUnittest
+
+var SubstringTests = TestSuite("SubstringTests")
+
+func checkMatch<S: Collection, T: Collection
+  where S.Index == T.Index, S.Iterator.Element == T.Iterator.Element,
+  S.Iterator.Element: Equatable>(
+  _ x: S, _ y: T, _ i: S.Index) {
+  
+  expectEqual(x[i], y[i])
+}
+
+let s = "abcdefg"
```

Comment:
Could you move this variable into each test that uses it?  This would make tests standalone.